File: app/src/main/java/com/boardgamegeek/sorter/PlayCountAscendingSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.boardgamegeek.R;

public class PlayCountAscendingSorter extends PlayCountSorter {
	public PlayCountAscendingSorter(@NonNull Context context) {
		super(context);
	}

	@StringRes
	@Override
	protected int getDescriptionId() {
		return R.string.collection_sort_play_count_asc;
	}

	@StringRes
	@Override
	public int getTypeResource() {
		return R.string.collection_sort_type_play_count_asc;
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/PlayCountAscendingSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import androidx.annotation.StringRes

import com.boardgamegeek.R
import com.boardgamegeek.entities.CollectionItemEntity

class PlayCountAscendingSorter(context: Context) : PlayCountSorter(context) {
    @StringRes
    public override val typeResId = R.string.collection_sort_type_play_count_asc

    @StringRes
    override val descriptionResId = R.string.collection_sort_play_count_asc

    override fun sort(items: Iterable<CollectionItemEntity>) = items.sortedBy { it.numberOfPlays }
}
